### ISSUE TITLE
What are the Darwin Core equivalents of Camtrap DP terms

### ISSUE THREAD

**Issue by peterdesmet** [tdwg/camtrap-dp#169]
As suggested in #167, we should indicate which Camtrap DP terms are borrowed from Darwin/Dublin Core using `rdfType`. For some terms, the relation is clear, for others less so. @ben-norton can you checkmark the ones you **consider equivalents** (appropriate mappings are not good enough, it needs to be the same concept). The rest we should discuss. Ping @dshorthouse

Clear

- [x] rightsHolder: http://purl.org/dc/terms/rightsHolder
- [x] bibliographicCitation: http://purl.org/dc/terms/bibliographicCitation
- [x] taxonID: http://rs.tdwg.org/dwc/terms/taxonID
- [x] scientificName: http://rs.tdwg.org/dwc/terms/scientificName
- [x] taxonRank: http://rs.tdwg.org/dwc/terms/taxonRank
- [x] vernacularNames description: http://rs.tdwg.org/dwc/terms/vernacularName
- [x] locationID: http://rs.tdwg.org/dwc/terms/locationID
- [x] longitude: http://rs.tdwg.org/dwc/terms/decimalLongitude
- [x] latitude: http://rs.tdwg.org/dwc/terms/decimalLatitude
- [x] coordinateUncertainty: http://rs.tdwg.org/dwc/terms/coordinateUncertaintyInMeters
- [x] habitat: http://rs.tdwg.org/dwc/terms/habitat
- [x] observationID: http://rs.tdwg.org/dwc/terms/occurrenceID
- [x] count: http://rs.tdwg.org/dwc/terms/individualCount
- [x] lifeStage: http://rs.tdwg.org/dwc/terms/lifeStage
- [x] sex: http://rs.tdwg.org/dwc/terms/sex
- [x] behaviour: http://rs.tdwg.org/dwc/terms/behavior
- [x] individualID: http://rs.tdwg.org/dwc/terms/organismID
- [x] classifiedBy: http://rs.tdwg.org/dwc/terms/identifiedBy
- [x] classificationTimestamp: http://rs.tdwg.org/dwc/terms/dateIdentified
- [x] observation remarks: http://rs.tdwg.org/dwc/terms/occurrenceRemarks

Less clear

- [x] ~taxonIDReference: http://rs.tdwg.org/dwc/terms/nameAccordingToID~ **not equivalent, because taxon ID reference (e.g. catalogueoflife.org) doesn't necessarily define the the taxon concept**
  - URL of the source (reference list) of the `taxonID`.
  - An identifier for the source in which the specific taxon concept circumscription is defined or implied. See nameAccordingTo.
- [x] locationName: http://rs.tdwg.org/dwc/terms/locality **yes, exact match**
  - Name given to the sampling location.
  - The specific description of the place. Less specific geographic information can be provided in other geographic terms (higherGeography, continent, country, stateProvince, county, municipality, waterBody, island, islandGroup). This term may contain information modified from the original to correct perceived errors or standardize the description.
- [x] ~classificationConfidence: http://rs.tdwg.org/dwc/terms/identificationVerificationStatus~ **not the same**
  - Accuracy confidence of the classification. Expressed as a probability, with `1` being maximum confidence. Provide an approximate value for human classifications.
  - A categorical indicator of the extent to which the taxonomic identification has been verified to be correct.
- [x] Do we consider a deployment the equivalent of a DwC:Event? If so, we could map:
  - deploymentID: http://rs.tdwg.org/dwc/terms/eventID **yes**
  - deployment remarks: http://rs.tdwg.org/dwc/terms/eventRemarks **yes**
- [x] Do we consider a sequence an event? I'd rather not. I'd also rather not consider deployment a parentEventID, with sequence and event. **Decided to consider both deployment and sequence an event, but not use parentEventID**
- [x] Do we consider observation timestamp equal to http://rs.tdwg.org/dwc/terms/eventDate? **yes**

**Comment by dshorthouse** [comment]
Not quite sure what to make of your `classificationConfidence`. Is the intent to:

- represent the determiner's own confidence in the identification ("I'm 90% confident that this is the identity of the organism that I said it is."),
- illustrate others' confidence in the determiner's identification ("The machine said this but we're merely 0.5 confident that it got it correct according to what we think is the identity of the organism."), 
- either/both of their confidence in the placement of a taxon in a classification ("We're 0.6 confident that this is the Family in which our organism identified to species is placed.")
 
I'm assuming it's more like the first or second one & not the last one. That last one illustrates that the use of the word 'classification' is probably best avoided because it's a loaded term. Either way, I see your term is a probability whereas `identificationVerificationStatus` is meant to be categorical. The first two items above for `classificationConfidence` illustrate that it's a slippery concept. I suppose you'd have to put yourself in the shoes of a user. What would a confidence of 0.65 vs 0.6 mean to a user? Who establishes the criteria for how to construct a value? Is it an algorithm or a human that constructs it? Would there be any loss of information to a user of the data if instead of `classificationConfidence` you used `identificationVerificationStatus` with categorical data or a mere boolean (i.e. "Yep, a human looked at it." vs "Nope, a human has not looked at it.")?

**Comment by peterdesmet** [comment]
@dshorthouse

1. It is the determiner (identified in `classifiedBy`: human or AI)'s own confidence in the identification. AI can specify this pretty accurately, for humans the percentage is more arbitrary, but it's generally not _set_ by the user. E.g. in the management software we have, we provide the AI value as is, but for humans we use `0.5` if they marked the boolean field `uncertain` and `1` if the classification was _verified_ by a human.
2. I personally think it's not a good equivalent of `identificationVerificationStatus`, if only that one is a percentage and the other is categorical, which are different concepts.
3. Regarding the word `classification`: it is a loaded term indeed and one mostly coming from the machine learning world, but with other connotations in biodiversity. Feedback welcome on the term in https://github.com/tdwg/camtrap-dp/issues/164

**Comment by dshorthouse** [comment]
> 1. It is the determiner (identified in `classifiedBy`: human or AI)'s own confidence in the identification. AI can specify this pretty accurately, for humans the percentage is more arbitrary, but it's generally not _set_ by the user. E.g. in the management software we have, we provide the AI value as is, but for humans we use `0.5` if they marked the boolean field `uncertain` and `1` if the classification was _verified_ by a human.

Aha! There *could* be two activities at play here, expressed in the same field. The first is if the original determiner self-declares their uncertainty and the second is if someone else indicates their agreement (or dissent?) with the determiner. I'm assuming _verification_ does not always mean agreement. What if there are several of the second kind, much like in iNaturalist? This does indeed get us into messy annotation space.

Regardless of whether or not it's an AI that made the original determination, these are distinct actions that attempt to convey trustworthiness. Should these be split such that downstream users are better informed when deciding what to toss or what to use? The question is if all this is merely noise or does it make the data more transparent and powerful? What does a user of the data expect to do when presented with any value in `classificationConfidence`?

**Comment by peterdesmet** [comment]
@dshorthouse see #170 to continue this discussion. Let's keep this one for Darwin Core equivalents.

**Comment by peterdesmet** [comment]
Action items:

- sequenceID = eventID
- deploymentID = parentEventID
- timestamp = eventDate
- deployment.comments = eventRemarks

**Comment by peterdesmet** [comment]
@timrobertson100 @gtucotuco If a deployment is the equivalent of a parentEvent, I assume it is then also an Event and can have Event terms? So even though a deployment has a `deploymentID` that is considered a http://rs.tdwg.org/dwc/terms/parentEventID, its comments can be considered http://rs.tdwg.org/dwc/terms/eventRemarks?

**Comment by tucotuco** [comment]
Yes, the deployment as an Event gets all the benefits of an Event. What is missing is the current vocabulary, but is badly needed, is and eventType.

**Comment by peterdesmet** [comment]
Thanks! And you’d suggest to rdfType then as:

- deploymentID: parentEventID
- sequenceID: eventID

Rather than eventID for both:

- deploymentID: eventID
- sequenceID: eventID

?

**Comment by tucotuco** [comment]
Deployment Event records would have deploymentID mapped to eventID (and
eventually an "eventType":"camera trap deployment". These could also have
parentEventIDs if you wanted to make an entire project into an Event, for
example.

Sequence Event records would have sequenceID mapped to eventID (and
eventually an "eventType":"image sequence"). It makes sense to me that each
of these would have a deployment Event as a parent.

**Comment by peterdesmet** [comment]
Ok, so reusing the same rdfType (here eventID) on different terms is not an issue?

**Comment by peterdesmet** [comment]
Oh, just realized, maybe I should only use `parentEventID` when it is used as a foreign key? So:

file |  field | rdfType
--- | --- | ---
deployments.csv | deploymentID | eventID
media.csv | mediaID | -
media.csv | deploymentID | parentEventID
media.csv | sequenceID | eventID
observations.csv | observationID | occurrenceID
observations.csv | deploymentID | parentEventID
observations.csv | sequenceID | eventID
observations | multimediaID | -

**Comment by tucotuco** [comment]
I am not sure what you are proposing here. The rdf:type should have an IRI to a class, not a property, so in this case all of them you populated in that column would be http://rs.tdwg.org/dwc/terms/Event. Otherwise I misunderstand completely what you are trying to do.

**Comment by peterdesmet** [comment]
Update: I discussed this with @tucotuco over a call.

1. We should not use `rdfType`: that is reserved for **classes** and Darwin Core terms are not classes. It is better to use [SKOS](https://www.w3.org/TR/2009/NOTE-skos-primer-20090818/) `skos:exactMatch`, `skos:narrowMatch`, `skos:broadMatch` to indicate this type of equivalents. I have now done so in #171 That PR is ready for review.
2. I have used `skos:exactMatch` for all the terms that are listed as "clear" in the issue body, as well as the ones under "less clear" that are now resolved.
3. Only two terms use `skos:narrowMatch`: `deploymentID` and `sequenceID` are narrow implementations of `eventID` (because they are specific types of events). After some discussion with @tucotuco it seemed best to map both to `eventID` and not map `deploymentID` to `parentEventID` when it occurs together with a `sequenceID`.
